mlflow/pmdarima.py

```diff
@@ -0,0 +1,364 @@
+"""
+The ``mlflow.pmdarima`` module provides an API for logging and loading ``pmdarima`` models.
+This module exports univariate ``pmdarima`` models in the following formats:
+
+Pmdarima format
+    Serialized instance of a ``pmdarima`` model using pickle.
+:py:mod:`mlflow.pyfunc`
+    Produced for use by generic pyfunc-based deployment tools and for batch auditing
+    of historical forecasts.
+
+.. _Pmdarima:
+    http://alkaline-ml.com/pmdarima/
+"""
+import os
+import logging
+import pickle
+import warnings
+import pandas as pd
+import yaml
+
+import mlflow
+from mlflow import pyfunc
+from mlflow.exceptions import MlflowException
+from mlflow.models import Model, ModelInputExample
+from mlflow.models.model import MLMODEL_FILE_NAME
+from mlflow.models.signature import ModelSignature
+from mlflow.models.utils import _save_example
+from mlflow.tracking._model_registry import DEFAULT_AWAIT_MAX_SLEEP_SECONDS
+from mlflow.tracking.artifact_utils import _download_artifact_from_uri
+from mlflow.utils.model_utils import _get_flavor_configuration
+from mlflow.utils.file_utils import write_to
+from mlflow.utils.requirements_utils import _get_pinned_requirement
+from mlflow.utils.environment import (
+    _mlflow_conda_env,
+    _validate_env_arguments,
+    _CONDA_ENV_FILE_NAME,
+    _process_pip_requirements,
+    _process_conda_env,
+    _CONSTRAINTS_FILE_NAME,
+    _REQUIREMENTS_FILE_NAME,
+)
+from mlflow.utils.docstring_utils import format_docstring, LOG_MODEL_PARAM_DOCS
+
+
+FLAVOR_NAME = "pmdarima"
+_MODEL_BINARY_KEY = "data"
+_MODEL_BINARY_FILE_NAME = "model.pmd"
+_MODEL_TYPE_KEY = "model_type"
+
+_logger = logging.getLogger(__name__)
+
+
+def get_default_pip_requirements():
+    """
+    :return: A list of default pip requirements for MLflow Models produced by this flavor.
+             Calls to :func:`save_model()` and :func:`log_model()` produce a pip environment that,
+             at a minimum, contains these requirements.
+    """
+    return [_get_pinned_requirement("pmdarima")]
+
+
+def get_default_conda_env():
+    """
+    :return: The default Conda environment for MLflow Models produced by calls to
+             :func:`save_model()` and :func:`log_model()`
+    """
+    return _mlflow_conda_env(additional_pip_deps=get_default_pip_requirements())
+
+
+@format_docstring(LOG_MODEL_PARAM_DOCS.format(package_name=FLAVOR_NAME))
+def save_model(
```

Comment:
Can we add the `@experimental` decorator to the public methods in this module?

Reply:
added